app/scripts/controllers/deployments.js

```diff
@@ -28,119 +39,151 @@ angular.module('openshiftConsole')
-            type: "warning",
-            details: "The active filters are hiding all deployments."
-          };
-        }
-        else {
-          delete $scope.alerts["deployments"];
-        }
-      }
-
-      $scope.showEmptyMessage = function() {
-        if ($filter('hashSize')($scope.deploymentsByDeploymentConfig) === 0) {
-          return true;
-        }
+          updateFilterWarning();
 
-        if ($filter('hashSize')($scope.deploymentsByDeploymentConfig) === 1 && $scope.deploymentsByDeploymentConfig['']) {
-          return true;
-        }
+          if (!action) {
+            // Loading of the page that will create deploymentConfigDeploymentsInProgress structure, which will associate running deployment to his deploymentConfig.
+            $scope.deploymentConfigDeploymentsInProgress = DeploymentsService.associateRunningDeploymentToDeploymentConfig($scope.deploymentsByDeploymentConfig);
+          } else if (action === 'ADDED' || (action === 'MODIFIED' && ['New', 'Pending', 'Running'].indexOf($filter('deploymentStatus')(deployment)) > -1)) {
+            // When new deployment id instantiated/cloned, or in case of a retry, associate him to his deploymentConfig and add him into deploymentConfigDeploymentsInProgress structure.
+            $scope.deploymentConfigDeploymentsInProgress[deploymentConfigName] = $scope.deploymentConfigDeploymentsInProgress[deploymentConfigName] || {};
+            $scope.deploymentConfigDeploymentsInProgress[deploymentConfigName][deploymentName] = deployment;
+          } else if (action === 'MODIFIED') {
+            // After the deployment ends remove him from the deploymentConfigDeploymentsInProgress structure.
+            var status = $filter('deploymentStatus')(deployment);
+            if (status === "Complete" || status === "Failed"){
+              delete $scope.deploymentConfigDeploymentsInProgress[deploymentConfigName][deploymentName];
+            }
+          }
 
-        return false;
-      };
+          // Extract the causes from the encoded deployment config
+          if (deployment) {
+            if (action !== "DELETED") {
+              deployment.causes = $filter('deploymentCauses')(deployment);
+            }
+          }
+          else {
+            angular.forEach($scope.deployments, function(deployment) {
+              deployment.causes = $filter('deploymentCauses')(deployment);
+            });
+          }
 
-      LabelFilter.onActiveFiltersChanged(function(labelSelector) {
-        // trigger a digest loop
-        $scope.$apply(function() {
-          $scope.deploymentConfigs = labelSelector.select($scope.unfilteredDeploymentConfigs);
+          Logger.log("deployments (subscribe)", $scope.deployments);
+        }));
+
+        watches.push(DataService.watch({
+          group: "extensions",
+          resource: "replicasets"
+        }, context, function(replicaSets) {
+          $scope.unfilteredReplicaSets = replicaSets.by("metadata.name");
```

Comment:
add a TODO comment that this will need to be updated to only the replicasets that aren't part of a deployment